gumdrop_derive does not compile with minimal versions

The crate versions specified in gumdrop_drive's Cargo.toml file are too low.  It can't compile when using those versions.  To demonstrate, do
```
$ cargo update -Zminimal-versions
$ cargo check
```
on any project that uses gumpdrop.  It will fail with errors like this:
```
error[E0599]: no method named `get_ident` found for reference `&syn::Path` in the current scope
    --> /home/somers/.cargo/registry/src/github.com-1ecc6299db9ec823/gumdrop_derive-0.8.0/src/lib.rs:1039:52
     |
1039 |                     Meta::Path(path) => match path.get_ident() {
     |                                                    ^^^^^^^^^ help: there is an associated function with a similar name: `is_ident`
```

--- a/gumdrop_derive/Cargo.toml
+++ b/gumdrop_derive/Cargo.toml
@@ -24,4 +24,4 @@ default_expr = ["syn/full"]
 [dependencies]
 proc-macro2 = "1"
 quote = "1"
-syn = "1"
+syn = "1.0.3"

